### PR TITLE
[Bug fixed]Fix dice_loss errors

### PR DESCRIPTION
**Pull request by xiexinch**
Fix training errors  #411

### CODE REVIEW

**Comment by zhongqiu1245** [comment]
Hi
I used your modified dice_loss to test, the original error disappeared, but a new error appeared.
My data is only two types (background and people), in the VOC format
No error is reported when loss is the Cross_Entropy or LovaszLoss
It will appear when loss is DiceLoss:
"num_classes=(1 if num_classes == 2 else num_classes))
RuntimeError: Class values must be smaller than num_classes."

In the config file, I call DiceLoss like this：
                                            **loss_decode=dict(type='DiceLoss')**
Thank you!

**Comment by xiexinch** [comment]
> Hi
> I used your modified dice_loss to test, the original error disappeared, but a new error appeared.
> My data is only two types (background and people), in the VOC format.
> 
> No error is reported when loss is the Cross_Entropy or LovaszLoss
> It will appear when loss is DiceLoss:
> "num_classes=(1 if num_classes == 2 else num_classes))
> RuntimeError: Class values must be smaller than num_classes."
> 2021-03-13 22-09-26屏幕截图
> 
> In the config file, I call DiceLoss like this：
> **loss_decode=dict(type='DiceLoss')**
> Is this method right?
> Thank you!

Hi @zhongqiu1245  
This error occurs if num_classes == 2, I am trying to fix it, sorry..

**Comment by zhongqiu1245** [comment]
Thanks, it works perfectly
Sorry to interrupt you, I want to try to change dice loss to tversky loss, but I am having trouble calculating the false positives and false negatives on the denominator.
How do you calculate false positives and false negatives?
Why is the denominator of your dice loss the sum of squares? Is there any benefit to doing this?
Thank you

File: mmseg/models/losses/dice_loss.py [hunk 1]
```diff
@@ -69,28 +69,38 @@ class DiceLoss(nn.Module):
...
-        else:
-            self.cls_criterion = binary_dice_loss
+        # assert loss_type in ['multi_class', 'binary']
+        # if loss_type == 'multi_class':
+        #     # self.cls_criterion = dice_loss
+        #     self.binary_cls = False
+        # else:
+        #     # self.cls_criterion = binary_dice_loss
+        #     self.binary_cls = True
+        # self.cls_criterion = dice_loss
```
Comment (xvjiarui) [inline mmseg/models/losses/dice_loss.py:90]: Why comment these lines?

Comment (xiexinch) [reply mmseg/models/losses/dice_loss.py:90]: Cause `dice_loss()` method has handled these two cases, I think it's no need to add extra codes.

Comment (xvjiarui) [reply mmseg/models/losses/dice_loss.py:90]: In this case, we could delete this code.

**Comment by xvjiarui** [comment]
Kindly ping @xiexinch

**Comment by xiexinch** [comment]
> Kindly ping @xiexinch

I will update it soon, sorry..

**Review by xvjiarui — APPROVED** [review]

**Comment by ChenLiufeng** [comment]
> > Hi
> > I used your modified dice_loss to test, the original error disappeared, but a new error appeared.
> > My data is only two types (background and people), in the VOC format.
> > No error is reported when loss is the Cross_Entropy or LovaszLoss
> > It will appear when loss is DiceLoss:
> > "num_classes=(1 if num_classes == 2 else num_classes))
> > RuntimeError: Class values must be smaller than num_classes."
> > 2021-03-13 22-09-26屏幕截图
> > In the config file, I call DiceLoss like this：
> > **loss_decode=dict(type='DiceLoss')**
> > Is this method right?
> > Thank you!
> 
> Hi @zhongqiu1245 This error occurs if num_classes == 2, I am trying to fix it, sorry..
@xiexinch I changed **loss_decode=dict(type='FocalLoss')**. And my num_classes=2. I met the same problem. 
J$0C}0 4K$NR)CZ)MV}C@2U
Could you fix it? Thanks!

**Pull request merged**